### ISSUE TITLE
Undestructible GorgonText in 2.0

### ISSUE THREAD

**Issue by AMartyn** [Tape-Worm/Gorgon#3]
TL/DR version:

   GC can't collect unused instances of GorgonText.

Details:

   In its constructor, GorgonText subscribes to the font's FontChanged event: 
`_font.FontChanged += FontChanged;`
This effectively creates a reference in the font to this instance of GorgonText. Since the object never unsibscribes from this event, the reference remains there forever and **GC can't collect any of the GorgonText objects** while their font is alive. If you create a font at the startup of your game and reuse it all the time while the game runs, all the instances of GorgonText you've ever created with this font will remain in memory with all their internal StringBuilders and stuff.

**Comment by Tape-Worm** [comment]
Yep.  That sure would create an event leak.  

It should be easy enough to fix by unassigning the event in the Dispose method or using a WeakEventManager to bind the event.

Unfortunately, I don't have a lot of time these days to look into this, but if you're willing to make a PR for it, I can merge it.  Otherwise it'll likely have to wait for a bit until I have the time to dig into it.

**Comment by AMartyn** [comment]
PR is ready: https://github.com/Tape-Worm/Gorgon/pull/4

**Comment by Tape-Worm** [comment]
So, I'm a complete dumbass.  I told you to use Dispose and I really shouldn't have, and you, being the nice person you are made me a PR, which, unfortunately I couldn't use.

I did however make the time to fix the bug, and a new release (2.0.1) is now up.  Again, I apologize, I should have been more careful.

I've also included a test repo project for this bug in the solution to ensure that the issue is solved.

**Comment by AMartyn** [comment]
Wow, you're cool. I've barely found time to make a straightforward modification in a couple of files, while you've managed to do it in a proper way all around! That's a great job, thank you a lot!
